ecosystem/sep-0031.md

```diff
@@ -65,21 +65,24 @@ This protocol involves the transfer of value, and so HTTPS is required for all e
 ### User Onboarding
 
 1. User onboarding happens with the sending anchor, and is out of the scope of this spec.
-1. Sending anchor should use this to collect any KYC about the receiver it will need so it doesn't have to do it at every transaction.
+2. The sending anchor should, at minimum, collect the KYC information of the sending and receiving users specified by the receiving anchor's [SEP-12](sep-0012.md) implementation.
 
 ### Payment Flow
 
 1. Payment initiation happens in the sending anchors interface, and is also out of the scope of this spec.
 1. The sending client chooses a destination region.
 1. The sending anchor fetches the TOML file to find the DIRECT_PAYMENT_SERVER API root and [`/info`](#info) endpoint fields for the receiving anchor for this region.
-1. Using the fields fetched from /info, the sending anchor collects all needed information from the sending client.
-1. The sending anchor POSTs all that information to the [/send](#send) endpoint of the receiving anchor.
-1. If information is missing, or the receiving anchor needs extra information to complete this transaction, `POST /send` will return `400 Customer Information Needed` along with the missing fields. The sending anchor should collect these and retry the entire `POST /send` call, until it succeeds.
+1. For every `*_sep12_type` value specified in the relevant asset's [`/info`](#info) object, the sending anchor makes a [`SEP-12 GET /customer`](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0012.md#customer-get) request.
+1. The sending anchor collects all information specified in the `GET /customer` response(s).
+1. The sending anchor PUTs all that information to the [`/customer`](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0012.md#customer-put) endpoint of the receiving anchor.
+1. The sending anchor POSTs the transaction information to the [/send](#send) endpoint of the receiving anchor.
+1. If information is missing, or the receiving anchor needs extra information to complete this transaction, `POST /send` will return `400 Customer Information Needed` or `400 Transaction Information Needed`.  The sending anchor should collect the necessary information and retry the entire `POST /send` call, until it succeeds.
```

Comment:
Its unclear what 'collect the necessary information and retry the entire post send call' means.  It makes sense for the transaction information, but for customer info do we need to do extra SEP12 PUTs, or do we pass it into the /send call?

I can imagine doing another SEP12 PUT with a new `type`, for example if we find out the amount is very large there might be a kyc_type `sep31-receiver-large-amount` and the sep12 implementation will know what info to collect for large amounts

Reply:
alternatively it might be fine to just put the fields needed directly here, and either ask you to send them to sep12, or to let you send them directly to /send

Reply:
The instruction is intentionally vague because I didn't want to get into the differences between the two 400 statuses in this section. I may just take this instruction out since its for the failure case anyway and just describe it in detail below.

Regarding your second comment, the client-side developer in me likes only needing 1 request for KYC and transaction info, but I think including parameters described in SEP-12 in a SEP-31 request is confusing and disorganized. I don't think its too much overhead to make 2 requests.